src/components/shared/Fields/Checkbox/HookForm/Checkbox.tsx

```diff
@@ -84,7 +83,10 @@ const HookFormCheckbox = ({
   );
 
   return (
-    <div className={classNames}>
+    <fieldset
+      className={classNames(mainClasses, className)}
+      disabled={disabled}
+    >
```

Comment:
When you set an input as disabled, the hook-form sets its value to undefined. Per docs (https://react-hook-form.com/api/useform/register/):
> disabled input will result in an undefined form value. If you want to prevent users from updating the input, you can use readOnly or disable the entire <fieldset />

readOnly doesn't work for checkboxes, so I went with wrapping the checkbox in a fieldset